File: DSA/Manav-Rajput/Day13_question_2.java

public class PalindromeChecker {

    public static boolean isPalindrome(int number) {
        String numStr = Integer.toString(number);
        return isPalindromeRecursive(numStr, 0, numStr.length() - 1);
    }

    private static boolean isPalindromeRecursive(String str, int left, int right) {
        if (left >= right) {
            return true;
        }
   
        if (str.charAt(left) != str.charAt(right)) {
            return false;
        }
        
        return isPalindromeRecursive(str, left + 1, right - 1);
    }

    public static void main(String[] args) {
        int number = 121;
        
        if (isPalindrome(number)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }
}
 

File: DSA/Ansh-Gulati/Day-13/Day-13_Question-2.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindromeHelper(const string &numberStr, int leftIndex, int rightIndex)
{
    if (leftIndex >= rightIndex)
    {
        return true;
    }
    if (numberStr[leftIndex] != numberStr[rightIndex])
    {
        return false;
    }
    return isPalindromeHelper(numberStr, leftIndex + 1, rightIndex - 1);
}

string isPalindrome(int number)
{
    string numberStr = to_string(number);

    if (isPalindromeHelper(numberStr, 0, numberStr.length() - 1))
    {
        return "YES";
    }
    else
    {
        return "NO";
    }
}

int main()
{
    int testNumber;
    cout << "Enter an integer: ";
    cin >> testNumber;

    cout << isPalindrome(testNumber) << endl;

    return 0;
}
